tfs/src/main/java/hudson/plugins/tfs/rm/ReleaseManagementHttpClient.java

```diff
@@ -57,6 +64,42 @@ public ReleaseArtifactVersionsResponse GetVersions(String project, List<Artifact
         String response = this.ExecutePostmethod(url, body);
         return new Gson().fromJson(response, ReleaseArtifactVersionsResponse.class);
     }
+
+    public void GetReleaseLogs(String project, String releaseId, BuildListener listener, FilePath filePath) throws ReleaseManagementException
+    {
+        String url = this.accountUrl + project + "/_apis/release/releases/" + releaseId + "/logs?api-version=4.0-preview.2";
+        InputStream inputStream = this.ExecuteGetMethodAsStream(url);
+
+        try (OutputStream out = filePath.write()) {
+            IOUtils.copy(inputStream, out);
+        } catch (Exception e) {
+            listener.getLogger().printf(e.getMessage());
+        }
+        return;
+    }
+
+    public String GetReleaseStatus(String project, String releaseId) throws ReleaseManagementException
+    {
+        String url = this.accountUrl + project + "/_apis/release/releases/" + releaseId + "?api-version=4.0-preview.4";
+        String response = this.ExecuteGetMethod(url);
+        ReleaseDetails releaseDetails = new Gson().fromJson(response, ReleaseDetails.class);
+        for (Map<String, Object> env : releaseDetails.getEnvironments())
+        {
+            if (((String) env.get("status")).equals("inProgress"))
+            {
+                return "inProgress";
+            }
+        }
+        return "finished";
+    }
+
+    public List<Project> GetProjectItems() throws ReleaseManagementException
+    {
+        String url = this.accountUrl + "/_apis/projects?api-version=1.0";
+        String response = this.ExecuteGetMethod(url);
+        TeamProject teamProject = new Gson().fromJson(response, TeamProject.class);
```

Comment:
Can we do something like, 
values = new JSONObject(response).getString("value");
List<Project> projects = new Gson().fromJson(values, TeamProject[].class);

The teamproject object is not really representing the teamproject. The api returns a list of teamproject and a count. Looks like we want an intermediate object to represent that, but the final outcome we are looking is List<Project>. Looks like just parsing the value field is good enough.